devfile/devworkspace-operator
# Add requirements for working with OLM

### What does this PR do?
* Add new kustomize target, `deploy/templates/olm` for generating files to be used in the bundle image
* Add Makefile rules for generating bundle manifests, bundle and index images
* Add catalogsource, etc. for testing catalogsource locally.

Todo:
* Remove remaining placeholders in base CSV and fill out description, etc.
* Test latest changes on OpenShift -- I can't get crc to run today
* Update README.md, etc. for changes
* Set up some automation for building/pushing bundle/index
* Update index build to include `--from-index`
* Prep everything for CPaaS (@JPinkney your help would be much appreciated!)

### What issues does this PR fix or reference?
Part of https://github.com/devfile/devworkspace-operator/issues/407

### Is it tested? How?
Test using added makefile rules:
- export env vars `DWO_BUNDLE_IMG` and `DWO_INDEX_IMG` for test
- `make generate_olm_bundle_yaml` - generate `deploy/bundle`
- `make generate_olm_bundle_image` - generate operator bundle image
- `make generate_olm_index_image` - generate operator index image
- `make register_catalogsource` - register catalog source in cluster
- Install operator via OperatorHub.

To test installing DWO as a dependency of WTO, you *should* be able to just apply the catalogsource
```yaml
cat <<EOF | oc apply -f -
apiVersion: operators.coreos.com/v1alpha1
kind: CatalogSource
metadata:
  name: custom-web-terminal-catalog
  namespace: openshift-marketplace
spec:
  sourceType: grpc
  image: quay.io/amisevsk/web-terminal-operator-index:dev
  publisher: amisevsk
  displayName: Web Terminal Operator Catalog
EOF
```
and install WTO from that catalog. I'll open a PR in WTO with those changes tomorrow.
